test/service/test_users.py

```diff
@@ -0,0 +1,205 @@
+            users.authenticate_via_geoaxis('test-token')
+        
+    def test_throws_on_dbase_error(self):
+        self.mock_requests.get(self.userprofile_addr, text='{"uid":"test-id","username":"test-name"}', status_code=201)
+        self.mock_insert_update = self.create_mock('bfapi.db.users.insert_or_update_user')
+        self.mock_insert_update.side_effect = helpers.create_database_error()
+        with self.assertRaises(DatabaseError):
+            users.authenticate_via_geoaxis('test-token')
+        
+    def test_successful_run(self):
+        self.mock_requests.get(self.userprofile_addr, text='{"uid":"test-id","username":"test-name"}', status_code=201)
+        self.mock_insert_update = self.create_mock('bfapi.db.users.insert_or_update_user')
+        self.mock_get_by_id = self.create_mock('bfapi.service.users.get_by_id')
+        inp_user = users.User(geoaxis_uid = "test-uid", user_name = "test-user_name")
+        self.mock_get_by_id.return_value = inp_user
+        outp_user = users.authenticate_via_geoaxis('test-token')
+        self.assertIsInstance(outp_user, users.User)
+        self.assertEqual(inp_user, outp_user)
+
+class AuthenticateViaApiKeyTest(unittest.TestCase):
+    def setUp(self):
+        self._mockdb = helpers.mock_database()
+        self._logger = logging.getLogger('bfapi.service.users')
+        self._logger.disabled = True
+        self.mock_get_user_by_api_key = self.create_mock('bfapi.db.users.select_user_by_api_key')
+
+    def tearDown(self):
+        self._mockdb.destroy()
+        self._logger.disabled = False
+        
+    def create_mock(self, target_name):
+        patcher = patch(target_name)
+        self.addCleanup(patcher.stop)
+        return patcher.start()
+        
+    def test_throws_on_db_error(self):
+        self.mock_get_user_by_api_key.side_effect = helpers.create_database_error()
+        with self.assertRaises(DatabaseError):
+            users.authenticate_via_api_key('test-api_key')
+        
+    def test_throws_on_failed_auth(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = None
+        with self.assertRaises(users.BeachfrontUnauthorized):
+            new_user = users.authenticate_via_api_key('test-api_key')
+        
+    def test_verifies_api_key(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = create_user_db_record()
+        with self.assertRaises(users.DatabaseMismatchError):
+            new_user = users.authenticate_via_api_key('not-the-returned-api_key')
+        
+    def test_return_good_user_id(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = create_user_db_record(api_key='test-api_key')
+        new_user = users.authenticate_via_api_key('test-api_key')
+        self.assertIsInstance(new_user, users.User)
+        self.assertEqual('test-user-id', new_user.geoaxis_uid)
+
+    def test_return_good_user_name(self):
```

Comment:
Can you rename this to `test_assigns_correct_*`?

Ref:
https://github.com/venicegeo/bf-api/blob/master/test/service/test_productlines.py#L412